### PR TITLE
Add models

### LINKED ISSUE #8: Add Models for Item, List, Category
- add typescript models for Item, List and Category 
- use the models throughout the application

### PR DESCRIPTION
**Pull request by nilsalm**
Should resolve #8

### CODE REVIEW

File: src/lib/util.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const deepClone = (obj: any) => {
+	return JSON.parse(JSON.stringify(obj));
+};
```
Comment (brokelyn) [inline src/lib/util.ts:3]: nice to put this into a utils file!

File: src/routes/lists/+page.server.ts [hunk 2]
```diff
@@ -7,19 +8,7 @@ export const load = ({ locals }) => {
...
-					})
-				)
-			) as Array<{
-				id: string;
-				name: string;
-				isTemplate: boolean;
-				user: string;
-			}>;
-			return lists;
+			return getListsQuery();
```
Comment (brokelyn) [inline src/routes/lists/+page.server.ts:11]: Nice! We should do more of refactoring db function into specific files!

**Review by brokelyn — CHANGES_REQUESTED** [review]
I really like the changes here!

The only thing that I dont get/wanna discuss is having a `Base` type and then adding an `id` to it?

why no adding the id directly to it? Is the `id` only used for the db or does it fulfill another purpose? 
I usually would say a `Base` type is useful if you have different subtypes, e.g. `BaseBall` is the base type of a `Football`.
Let me know what you think.


Another thing. I really like putting the db functions in the `pocketbase` file. At some point we might wanna split the file into multiple with different functions :rocket:

File: src/routes/lists/+page.server.ts (hunk 2 above)
Comment (nilsalm) [reply src/routes/lists/+page.server.ts:11]: Do mean like a query file or different files for like ListQueries, ItemQueries, etc etc?

File: src/lib/models/Item.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import type { List } from './List';
+import type { Category } from './Category';
+
+export type BaseItem = {
+	name: string;
+	quantity: number;
+	picked: boolean;
+	category: Category | string;
+	list: List | string;
+	user: User | string;
```
[the remark this answers is not in the document]
Comment (nilsalm) [reply src/lib/models/Item.ts:11]: Very true, should have updated this file! good catch!

**Comment by nilsalm** [comment]
I followed a pattern that i found [over here](https://auth0.com/blog/node-js-and-typescript-tutorial-build-a-crud-api/) where the `BaseType` is used for POST actions where you are creating new things that dont have an id yet, and the `Type` woul be used when querying things. But I do agree that that's a bit overcomplicated for this. I'll simplify!

**Review by brokelyn — APPROVED** [review]
Nice looks good now! :rocket:

**Pull request merged**